### ISSUE TITLE
OData .NET Client fails to read instance annotations of complex types.

### ISSUE THREAD

**Issue by VirusQuartirus** [OData/odata.net#1198]
*ODataLib 6.x allows to create and consume a payload with instance annotations of complex types. However in ODataLib 7.x this functionality is broken.
Also there is no more possibility to write complex type annotations in ODataLib 7.x based service.*

### Assemblies affected

*OData .Net lib 7.4.*

### Reproduce steps

*Try to read response like this:
```
{
    "error": {
        "code": "36",
        "message": "Bad stuff",
        "innererror": {
            "message": "Exception message",
            "type": "",
            "stacktrace": ""
        },
        "@osa.fault": {
            "@odata.type": "#WebApplication16.Models.Fault",
            "Bar": 42,
            "Foo": {
                "Foo": "Foo",
                "Bar": "Bar"
            }
        }
    }
}
```
*

### Expected result

*In ODataLib 6.x this would return ODataError with single instance annotation.*

### Actual result

*Microsoft.OData.ODataException: 'An internal error 'ODataJsonLightPropertyAndValueDeserializer_ReadPropertyValue' occurred.'*

**Comment by mikepizzo** [comment]
This is likely a result of making complex types into resources in ODL 7.x.

InstanceAnnotation.Value is an ODataValue, and complex types are represented as a Resource (which is not an ODataValue).

**Comment by VirusQuartirus** [comment]
I see... But why?
I mean that ODL 7.x is broken by design. It doesn't support all features of ODL 6.x in proper way. It can't consume/produce a valid OData V4 payload.
Is it possible to fix this issue in ODL 7.x?
For instance by inheriting ODataResource from ODataValue instead of ODataItem?

**Comment by mikepizzo** [comment]
Yes; we can (and should) fix this.  But we are planning on shipping out 7.5 this week, and we won't have time to fix it for that release, so it will go in the next release (7.5.x or 7.6).

**Comment by korygin** [comment]
Any ETA on when a fix may be provided? We are experiencing the same issue after migrating to ODataLib 7.x.  Thanks!
